data_pipeline/twitter_ads_api/twitter_ads_api_pipeline.py

```diff
@@ -85,36 +84,87 @@ def get_param_dict_from_api_query_parameters(
     return result
 
 
+def get_final_end_date(
+    api_query_parameters_config: TwitterAdsApiApiQueryParametersConfig,
+    initial_start_date_value: datetime
+) -> datetime:
+    yesterday_date = get_yesterdays_date()
+    end_date_by_period = initial_start_date_value + timedelta(
+        days=api_query_parameters_config.parameter_values.ending_period_per_day
+    )
+    if end_date_by_period > yesterday_date:
+        final_end_date_value = yesterday_date
+    else:
+        final_end_date_value = end_date_by_period
+    return final_end_date_value
+
+
+def get_end_date_value_of_period(
+    start_date_value: datetime,
+    final_end_date_value: datetime,
+    days_in_period: int
+) -> datetime:
+    period_end_date_value = start_date_value + timedelta(days=days_in_period)
+    if period_end_date_value > final_end_date_value:
+        end_date_value = final_end_date_value
+    else:
+        end_date_value = period_end_date_value
+    return end_date_value
+
+
 def iter_bq_compatible_json_response_from_resource_with_provenance(
     source_config: TwitterAdsApiSourceConfig
 ) -> Any:
     if source_config.api_query_parameters:
         api_query_parameters_config = source_config.api_query_parameters
-        placement_value_list = api_query_parameters_config.parameter_values.placement_value
-        if api_query_parameters_config.parameter_names_for.placement:
-            assert placement_value_list
-            assert api_query_parameters_config.use_start_date_from_bigquery
-            dict_value_list_from_bq = list(iter_dict_from_bq_query_for_bigquery_source_config(
-                api_query_parameters_config.parameter_values.from_bigquery
-            ))
-            LOGGER.debug("dict_value_list_from_bq: %r", dict_value_list_from_bq)
-            for dict_value_from_bq in dict_value_list_from_bq:
+        dict_value_list_from_bq = list(iter_dict_from_bq_query_for_bigquery_source_config(
+            api_query_parameters_config.parameter_values.from_bigquery
+        ))
+        LOGGER.debug("dict_value_list_from_bq: %r", dict_value_list_from_bq)
+        for dict_value_from_bq in dict_value_list_from_bq:
+            entity_id_from_bq = dict_value_from_bq['entity_id']
+            start_date_str_from_bq = dict_value_from_bq['start_date']
+            initial_start_date_value = (
+                datetime.strptime(start_date_str_from_bq, '%Y-%m-%d').date()
+            )
+            if initial_start_date_value <= datetime.strptime('2015-01-01', '%Y-%m-%d').date():
```

Comment:
Is the hardcoded date intentional?

Reply:
`min_api_start_date`

Reply:
added filter to SQL query instead of using this variable